In [38]:
import pandas as pd

In [39]:
df = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv')
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


неизвестно около 20 % возрастов всех пассажиров и 77 % палуб

In [15]:
1 - 203 / 891

0.7721661054994389

In [6]:
df.sex.value_counts()

male      577
female    314
Name: sex, dtype: int64

In [9]:
all_pas = df.shape[0]

mask = (df['adult_male'])
old_men = df[mask].shape[0]

print(f"old men: {old_men}, all passengers: {all_pas}")
print("persent of old men:", round(old_men / all_pas * 100, 3))

old men: 537, all passengers: 891
persent of old men: 60.269


In [29]:
df.pclass.value_counts()

3    491
1    216
2    184
Name: pclass, dtype: int64

In [30]:
df.pclass.value_counts(normalize=True)

3    0.551066
1    0.242424
2    0.206510
Name: pclass, dtype: float64

В первом классе - 216 человек или 24%,
во втором - 184 человека или 21%
и в тертьем большинство - 491 человек или 55%

In [31]:
df.describe(include='all')

survived      pclass   sex         age       sibsp       parch  \
count   891.000000  891.000000   891  714.000000  891.000000  891.000000   
unique         NaN         NaN     2         NaN         NaN         NaN   
top            NaN         NaN  male         NaN         NaN         NaN   
freq           NaN         NaN   577         NaN         NaN         NaN   
mean      0.383838    2.308642   NaN   29.699118    0.523008    0.381594   
std       0.486592    0.836071   NaN   14.526497    1.102743    0.806057   
min       0.000000    1.000000   NaN    0.420000    0.000000    0.000000   
25%       0.000000    2.000000   NaN   20.125000    0.000000    0.000000   
50%       0.000000    3.000000   NaN   28.000000    0.000000    0.000000   
75%       1.000000    3.000000   NaN   38.000000    1.000000    0.000000   
max       1.000000    3.000000   NaN   80.000000    8.000000    6.000000   

              fare embarked  class  who adult_male deck  embark_town alive  \
count   891.000000      889    891  891        891  203          889   891   
unique         NaN        3      3    3          2    7            3     2   
top            NaN        S  Third  man       True    C  Southampton    no   
freq           NaN      644    491  537        537   59          644   549   
mean     32.204208      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
std      49.693429      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
min       0.000000      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
25%       7.910400      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
50%      14.454200      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
75%      31.000000      NaN    NaN  NaN        NaN  NaN          NaN   NaN   
max     512.329200      NaN    NaN  NaN        NaN  NaN          NaN   NaN   

       alone  
count    891  
unique     2  
top     True  
freq     537  
mean     NaN  
std      NaN  
min      NaN  
25%      NaN  
50%      NaN  
75%      NaN  
max      NaN

Из этой таблицы можем сделать несколько выводов:
1. всего 891 пассажиров - 577 мужчин и 314 женщин
2. 38 % выжило
3. средний возраст - 30 лет
4. самому старому человеку 80 лет, а младшему еще нет года
5. пассажиры жили в 147 разных номерах
6. большинство(644 человека) сели на борт в порту S (Southampton)
7. средняя стоимость билета 32$ 
8. самый дешевый билет оказался бесплатным, а самый дорогой стоит 512$

In [11]:
survived = df[df["survived"] == 1]

all_survived = survived.shape[0]
mask = (survived['adult_male'])
old_men = survived[mask].shape[0]

print(f"survived old men: {old_men}, all survived passengers: {all_survived}")
print("persent of survived old men:", round(old_men / all_survived * 100, 3))

survived.head()

survived old men: 88, all survived passengers: 342
persent of survived old men: 25.731


survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
1         1       1  female  38.0      1      0  71.2833        C   First   
2         1       3  female  26.0      0      0   7.9250        S   Third   
3         1       1  female  35.0      1      0  53.1000        S   First   
8         1       3  female  27.0      0      2  11.1333        S   Third   
9         1       2  female  14.0      1      0  30.0708        C  Second   

     who  adult_male deck  embark_town alive  alone  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
8  woman       False  NaN  Southampton   yes  False  
9  child       False  NaN    Cherbourg   yes  False

процент взрослых мужчин среди выживших уменьшился в 2.5 раза


2.4

In [47]:
survived.describe(include='all')

survived      pclass     sex         age       sibsp       parch  \
count      342.0  342.000000     342  290.000000  342.000000  342.000000   
unique       NaN         NaN       2         NaN         NaN         NaN   
top          NaN         NaN  female         NaN         NaN         NaN   
freq         NaN         NaN     233         NaN         NaN         NaN   
mean         1.0    1.950292     NaN   28.343690    0.473684    0.464912   
std          0.0    0.863321     NaN   14.950952    0.708688    0.771712   
min          1.0    1.000000     NaN    0.420000    0.000000    0.000000   
25%          1.0    1.000000     NaN   19.000000    0.000000    0.000000   
50%          1.0    2.000000     NaN   28.000000    0.000000    0.000000   
75%          1.0    3.000000     NaN   36.000000    1.000000    1.000000   
max          1.0    3.000000     NaN   80.000000    4.000000    5.000000   

              fare embarked  class    who adult_male deck  embark_town alive  \
count   342.000000      340    342    342        342  136          340   342   
unique         NaN        3      3      3          2    7            3     1   
top            NaN        S  First  woman      False    C  Southampton   yes   
freq           NaN      217    136    205        254   35          217   342   
mean     48.395408      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
std      66.596998      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
min       0.000000      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
25%      12.475000      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
50%      26.000000      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
75%      57.000000      NaN    NaN    NaN        NaN  NaN          NaN   NaN   
max     512.329200      NaN    NaN    NaN        NaN  NaN          NaN   NaN   

        alone  
count     342  
unique      2  
top     False  
freq      179  
mean      NaN  
std       NaN  
min       NaN  
25%       NaN  
50%       NaN  
75%       NaN  
max       NaN

из этих данных можно узнать что:
1. выжило всего 342 человека - 233 женщины(68%) и 109 мужчин(38%), то есть осталось в живых (233/314 * 100) 74% женщин и только (109/577 * 100) 19% мужчин
2. средний возраст почти не отличается от исходного - 28 лет
3. средняя стоимость билета выросла на 16$, а это значит, что больше шансов выжить с дорогим билетом
